### PR TITLE
trivial: Allow EFI application path to come from a different root

### PR DESCRIPTION
**Pull request by superm1**
I noticed this was a hardcoded path.

### CODE REVIEW

**Comment by hughsie** [comment]
If it works with CI, LGTM, thanks.

**Comment by superm1** [comment]
That fixes up CI, you OK if I squash that, or do you want a different approach?

File: src/fu-common.c [hunk 1]
```diff
@@ -850,6 +850,7 @@ fu_common_get_path (FuPathKind path_kind)
 		if (tmp != NULL)
 			return g_build_filename (tmp, DATADIR, PACKAGE_NAME, NULL);
 		return g_build_filename (DATADIR, PACKAGE_NAME, NULL);
+#ifdef EFI_APP_LOCATION
 	/* /usr/libexec/fwupd/efi */
 	case FU_PATH_KIND_EFIAPPDIR:
 		tmp = g_getenv ("FWUPD_EFIAPPDIR");
```
Comment (hughsie) [inline src/fu-common.c]: I'd move the #ifdef here, and have an #else that just returns NULL.

File: src/fu-common.h [hunk 2]
```diff
@@ -19,7 +19,9 @@ typedef enum {
 typedef enum {
 	FU_PATH_KIND_CACHEDIR_PKG,
 	FU_PATH_KIND_DATADIR_PKG,
+#ifdef EFI_APP_LOCATION
```
Comment (hughsie) [inline src/fu-common.h]: If you do this then it changes the enum ordering depending on the compile settings, which would be uncool to debug. Just drop these and do the #else thing pls. Then squash, then push. Thanks.

**Pull request merged**